### PR TITLE
DHFPROD-6418:Adding a new endpoint for updating steps

### PR DESCRIPTION
**Pull request by srinathgit**
### Description
BE:
- Adding a new endpoint for updating steps.
~~This PR shouldn't be merged yet. This needs changes to UI. I have opened it for review and for the tests to run. If all tests run fine, I will mark it as a draft. UI engineer will then use this PR to make the UI changes and open another PR~~

UI: 
- Creating a step with a duplicate name will now return an error message from the BE instead of overwriting the existing step.
- Updates to an existing step uses endpoint for updating steps
- Added e2e test coverage

#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Code passes ESLint tests
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

**Review by rjrudin — DISMISSED** [review]
Looks good

**Review by ryanjdew — DISMISSED** [review]

**Comment by briantang** [comment]
Completed the UI functionality/tests for this bug and squashed with @srinathgit 's backend work. Should be ready for review now.

**Review by Sanjeevani19 — COMMENTED** [review]
@briantang The functionality works as expected, few css changes are needed when adding a step with same name in mapping, Please look into the screenshot attached, The text alignment seems to be bit off

Screen Shot 2021-01-13 at 4 37 52 PM


Also there are some linting failures in the pipeline.

**Review by wooldridge — CHANGES_REQUESTED** [review]
I'm seeing the following unit test failures:
```
 FAIL  src/components/entities/create-edit-step/create-edit-step.test.tsx (8.569s)
  Create Edit Step Dialog component
    ✓ Verify Edit Merging dialog renders correctly for a read only user (128ms)
    ✓ Verify New Merging Dialog renders  (60ms)
    ✓ Verify save button is always enabled (82ms)
    ✓ Verify Save button requires all mandatory fields (205ms)
    ✓ Verify able to type in input fields and typeahead search in collections field (271ms)
    ✓ Verify able to type in input fields and typeahead search in collections field (265ms)
    ✕ Verify Edit Merging dialog renders correctly (113ms)

  ● Create Edit Step Dialog component › Verify Edit Merging dialog renders correctly

    expect(jest.fn()).toBeCalledWith(...expected)

    Expected: {"collection": "matchCustomers", "description": "merge customer description", "name": "mergeCustomers", "selectedSource": "collection", "sourceQuery": "cts.collectionQuery(['matchCustomers'])", "targetEntityType": "Customer", "timestamp": "/envelope/headers/createdOn"}

    Number of calls: 0

      225 | 
      226 |     fireEvent.click(getByText("Save"));
    > 227 |     expect(data.editMerging.createStepArtifact).toBeCalledWith({
          |                                                 ^
      228 |       name: "mergeCustomers",
      229 |       targetEntityType: "Customer",
      230 |       description: "merge customer description",

      at Object.<anonymous> (src/components/entities/create-edit-step/create-edit-step.test.tsx:227:49)

  ● Create Edit Step Dialog component › Verify Edit Merging dialog renders correctly

    TypeError: props.updateStepArtifact is not a function

      206 |       props.createStepArtifact(getPayload());
      207 |     } else {
    > 208 |       props.updateStepArtifact(getPayload());
          |             ^
      209 |     }
      210 |     props.setOpenStepSettings(false);
      211 |     props.resetTabs();

      at handleSubmit (src/components/entities/create-edit-step/create-edit-step.tsx:208:13)
      at Button._this.handleClick (node_modules/antd/lib/button/button.js:158:9)
      at HTMLUnknownElement.callCallback (node_modules/react-dom/cjs/react-dom.development.js:188:14)
```

**Comment by briantang** [comment]
Thanks @wooldridge @Sanjeevani19. Addressed CSS alignment, linting, and tests.

File: marklogic-data-hub-central/ui/e2e/cypress/integration/curation/curate/mapping.spec.tsx [hunk 1]
```diff
@@ -119,6 +127,17 @@ describe("Mapping", () => {
...
+    //verify mapping step with duplicate name cannot be created
+    cy.waitUntil(() => curatePage.addNewStep().click());
+    createEditMappingDialog.setMappingName(mapStep);
+    createEditMappingDialog.setSourceRadio("Query");
+    createEditMappingDialog.setQueryInput("test");
+    createEditMappingDialog.saveButton().click({force: true});
+    //error message should be displayed instead of step details auto open
+    cy.findByLabelText(`${mapStep}-details-header`).should("not.be.visible");
+    loadPage.duplicateStepErrorMessage();
+    loadPage.confirmationOptions("OK").click();
```
Comment (ngodugu-marklogic) [inline marklogic-data-hub-central/ui/e2e/cypress/integration/curation/curate/mapping.spec.tsx:139]: Can we add similar validations for Match and Merge Steps. And once we click "OK" can we also check whether the modal is not visible anymore.

Comment (briantang) [reply marklogic-data-hub-central/ui/e2e/cypress/integration/curation/curate/mapping.spec.tsx:140]: Thanks @ngodugu-marklogic , I've added the same checks for Match and Merge and also added a check that the modal is no longer visible afterwards

**Review by ngodugu-marklogic — APPROVED** [review]

**Review by Sanjeevani19 — APPROVED** [review]

**Review by wooldridge — APPROVED** [review]
Looks good, @briantang! All unit tests pass now. Manual tests of duplicate names for all four step types work as expected, with error dialog appearing and no duplicate steps created.

**Pull request merged**